src/olympia/files/templates/files/node.html

```diff
@@ -1,7 +1,12 @@
+{% if value['diff'] %}
+    {%- set files_url = value['url'] %}
+{% else %}
+    {%- set files_url = url('files.list', value['id'], 'file', value['short']) %}
```

Comment:
Not really a fan of these magic `value[...]` variables.  If `value['diff']` then `value['url']` is a compare link? 😕 But it's not otherwise?

Reply:
Neither am I. The `node.html` is being used in a template-tag that get's used for compare and regular browse views, thus it behaves differently based on context.

I could move the url resolving from the file-viewer to this file but the file-viewer uses a bit more context to resolve the compare-url as it needs the left and right files from the diff, and these aren't available in the `node.html` file, unfortunately. 

We can, however, use the URL from the file-viewer because for diffs the URL get's calculated after the caching, so this is always up to date.

This will certainly require a cleanup at some point but I tried to keep the diff and risk small.

Reply:
Can you add a variation of your second paragraph as a comment in this template? It took me a while to wrap my brain around why this was needed

Reply:
Added in f108379ed7. It's a bit verbose but I guess that's better than none.